typescript/api/api_service_client/api_client.ts

```diff
@@ -0,0 +1,143 @@
+//"io/ioutil"
+import { Result, ok, err } from "neverthrow";
+import * as httpStatusCode from "http-status-codes";
+import * as axios from "axios";
+
+const PERSON_ENDPOINT: string = "person";
+const TEXT_CONTEXT_TYPE: string = "text/plain"; //TODO (Ali)
+
+const TIMEOUT_SECONDS: number = 2; //TODO (Ali)
+const INCREMENT_BOOKS_READ_ENDPOINT: string = "incrementBooksRead";
+
+const HEALTHCHECK_URL_SLUG: string = "health";
+const HEALTHY_VALUE: string = "healthy";
+
+class Person {
+	private readonly booksRead: number;
+
+	constructor() {}
+}
+
+class APIClient {
+	//httpClient http.Client //TODO (Ali) - might not need since we have axios ;)
+	private readonly ipAddr: string;
+	private readonly port: number;
+	
+	constructor (ipAddr: string, port: number) {
+		this.ipAddr = ipAddr;
+		this.port = port;
+	}
+
+	public async addPerson(id: number): Promise<Result<null, Error>> {
+		const url: string = this.getPersonUrlForId(id);
+		const resp: axios.AxiosResponse<any> = await axios.default.post(url, null); //TODO (Ali) - might need to catch error to make up for line below ; content type missing in POST request
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
+		// 	return stacktrace.Propagate(err, "An error occurred making the request to add person with ID '%v'", id)
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("Adding person with ID '" + id +  "' returned non-OK status code " + resp.status + ""));
+		}
+		return ok(null);
+	}
+
+	public async getPerson(id: number): Promise<Result<Person, Error>> {
+		const url: string = this.getPersonUrlForId(id);
+		const resp: axios.AxiosResponse<any> = await axios.default.get(url); //TODO (Ali) - might need to catch error to make up for line below
+		
+		//TOOD (Ali) - since I removed http.Client struct, I might remove the following error check
+		// if err !== nil {
+		// 	return Person{}, stacktrace.Propagate(err, "An error occurred making the request to get person with ID '%v'", id)
+		// }
+		if (resp.status !== httpStatusCode.StatusCodes.OK) {
+			return err(new Error("Getting person with ID '" + id + "' returned non-OK status code " + resp.status + ""));
+		}
+		const body: any = resp.data; //TODO (Ali) - building this for JSON.parse
```

Comment:
Is this really `any` type? It's not `string`, or `bytes`?

Reply:
Oh wait, the documentation said it was <any> but then I found that online it said it returned a string: https://stackoverflow.com/questions/62290771/axios-returns-a-string-instead-of-json-vue-js